### ISSUE TITLE
Rule MD046’s `consistent` errors on documents with consistent block style

### ISSUE THREAD

**Issue by copperwalls** [markdownlint/markdownlint#221]
test.md:

    # I Believe This Document Should Pass

    ```sh
    $ echo "Aloha, world!"
    ```

    ## But It Triggers MD046

    ```sh
    $ echo "Hmm..."
    ```

**Comment by jaymzh** [comment]
I cannot repro this:

    [phil@vulturus (master) markdownlint]$ cat /tmp/test2.md 
    # I Believe This Document Should Pass
    
    ```sh
    $ echo "Aloha, world!"
    ```
    
    ## But It Triggers MD046
    
    ```sh
    $ echo "Hmm..."
    ```
    [phil@vulturus (master) markdownlint]$ mdl /tmp/test2.md 
    /tmp/test2.md:3: MD014 Dollar signs used before commands without showing output
    /tmp/test2.md:9: MD014 Dollar signs used before commands without showing output
    
    A detailed description of the rules is available at https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md
    [phil@vulturus (master) markdownlint]$

**Comment by jaymzh** [comment]
oh, I missed the part where this only happens on `consistent`. I can repro this now

**Comment by jaymzh** [comment]
OK, the attached PR makes this work for me.
